solutions/cses-1163.mdx

```diff
@@ -105,3 +107,111 @@ int main(){
 	return 0;
 }
 ```
+
+</CPPSection>
+<JavaSection>
+## Implementation
+[Unofficial Editorial (C++)](https://codeforces.com/blog/entry/83295)
+
+**Time Complexity:** $\mathcal{O}(n)$
+
```

Comment:
wait why is the editorial in a language section

Reply:
There's another editorial in another section, but it doesn't describe at all what I do in the example code. (They start from the beginning and calculate, while I take all the values and start calculating backwards).

Reply:
hm
then add a "Solution 2", i think that's how alternate implementations work

Reply:
There's already a Solution 2 that ~kinda~ touches on what I do but tbh it's pretty trash. Should I delete it and replace it with mine?

Reply:
trash? lemme look at it real quick

Reply:
Sure

Reply:
Also is it a problem that the java code is only available for one solution?

Reply:
no, not really

Reply:
alr here it is: https://paste.pythondiscord.com/ixajuliliy.cpp
it'd be cool if you could credit me in the authors

Reply:
Should I just put Sans Papyrus (or do you have a name)?

Reply:
yeah lmao
Kevin Sheng